Add Phel language

JesusValeraDev opened the pull request:
## 📚 Description

[Phel](https://phel-lang.org) is a functional Lisp dialect that compiles to PHP, drawing heavily from Clojure. This PR adds Phel language support to Linguist.

### Ecosystem

- Website: https://phel-lang.org
- Source: https://github.com/phel-lang/phel-lang (MIT)
- Featured projects: https://github.com/phel-lang/awesome-phel

### Editor / tooling support

- IntelliJ IDEA: https://github.com/phel-lang/phel-intellij-plugin · [Marketplace](https://plugins.jetbrains.com/plugin/28459-phel-lang/)
- VS Code: https://github.com/phel-lang/phel-vs-code-extension
- Vim: https://github.com/danirod/phel.vim
- Emacs: https://codeberg.org/mmontone/interactive-lang-tools

### Samples

Five focused samples (191 lines total) mirroring Linguist's Clojure convention of short, feature-distinct snippets:

| File | Demonstrates |
|---|---|
| `data-pipeline.phel` | `for [x :in xs]` / `:pairs` bindings, threading macros |
| `html-rendering.phel` | Hiccup-style HTML DSL via `phel.html` |
| `interfaces.phel` | `definterface`, `defstruct`, polymorphic dispatch |
| `macro-playground.phel` | `defmacro` with quasiquote / unquote-splicing |
| `php-integration.phel` | PHP interop (`php/->`, `php/::`, `php/new`, `php/sprintf`) |

All samples are sourced verbatim from [`phel-lang/docs/examples/`](https://github.com/phel-lang/phel-lang/tree/main/docs/examples) and have been executed against the current Phel CLI (v0.36.0-beta) — each exits 0 and produces the expected output.

### Syntax highlighting

`tm_scope: source.clojure` reuses the Clojure grammar intentionally — Phel's surface syntax is Clojure-compatible enough that Clojure highlighting gives a good baseline. A dedicated `source.phel` grammar may follow once the language has its own TextMate bundle.

## 📓 Checklist

- [x] **I am adding a new language.**
  - [x] The extension is in use on GitHub: https://github.com/search?q=path%3A*.phel&type=code
  - [x] Real-world usage samples are included under [`samples/Phel/`](https://github.com/phel-lang/linguist/tree/add-phel-language/samples/Phel).
    - Sample license: MIT (sourced from [phel-lang/phel-lang](https://github.com/phel-lang/phel-lang/blob/main/LICENSE))
  - [ ] Syntax highlighting grammar — reuses `source.clojure` (see note above).
  - [ ] Heuristics — not required, `.phel` is unique to this language.

JesusValeraDev committed:
Add Phel language

Chemaclass reviewed (approved):
Great idea! Highlighting phel files using common lisp would be a great improvement, especially people will be able to understand the code easier when reading it directly from GitHub 🧠💡

lildude commented on lib/linguist/languages.yml:
```diff
@@ -4771,6 +4771,15 @@ PEG.js:
+  codemirror_mime_type: text/x-common-lisp
+  color: "#512da8"
+  extensions:
+    - ".phel"
```
You've missed step 4 of adding a language in the CONTRIBUTING.md file as there should be a `language_id` here 😉

JesusValera replied:
Thank you so much for the review @lildude 😃

lildude reviewed (dismissed):
Popularity is not sufficient for inclusion yet, but that doesn't mean you can't address the identified issues in the mean time.

See inline comment.

Please remove the `core.phel` file. It's massive and waaaay too big.

Please also remove the `sample.phel` file as this is clearly a "Hello world" sample and not a real-world example.

JesusValeraDev committed:
Add language_id to Phel

JesusValeraDev committed:
Replace massive core.phel & sample.phel files by a snake game in Phel

JesusValeraDev committed:
Update Phel examples using Lisp syntax

JesusValeraDev committed:
Use Clojure syntax highlighting for Phel

Chemaclass committed:
Merge branch 'github-linguist:main' into add-phel-language

Chemaclass committed:
fix(samples): use dot namespace separator in Phel samples

Phel namespaces use '.' as separator, not PHP-style '\'. Sync samples
with canonical examples in phel-lang/docs/examples/.

Chemaclass committed:
feat(samples): add canonical Phel examples covering core language features

Adds nine samples synced from phel-lang/docs/examples/: basic types,
arithmetic, control flow, functions and recursion, data structures,
interfaces, async/concurrency, CLI, and transducers. Improves Bayesian
classifier coverage and showcases idiomatic Phel.

Chemaclass committed:
feat(phel): register phel CLI as interpreter

Aligns Phel entry with peer Lisp dialects (Hy, Fennel, Janet) which
declare their CLI binary as an interpreter.

Chemaclass committed:
ref(samples): trim Phel samples to focused feature set

Mirrors Clojure's approach: short snippets each showing one distinct
feature. Drops primer-level overlap (basic-types, arithmetic, control-flow,
functions-recursion) and oversized samples (data-structures 232 lines,
transducers 387 lines, cli, async-concurrency). Keeps five samples
covering PHP interop, macros, hiccup DSL, for-comprehensions, and
structs/interfaces. 191 lines total across 5 files.

Chemaclass commented:
Quick adoption update since this PR was opened (Feb 2023):

**Core project (May 2026)**
- [phel-lang/phel-lang](https://github.com/phel-lang/phel-lang) — 505 stars, 29 forks, 25 contributors
- Active development: v0.39.0 released today; 5 releases in the past three weeks
- Created 2020-04-19, continuous activity for 6 years

**Ecosystem (official)**
- IntelliJ plugin — https://github.com/phel-lang/phel-intellij-plugin
- VS Code extension — https://github.com/phel-lang/phel-vs-code-extension
- Libraries: `phel-pdo`, `phel-sql`, `phel-log`, `phel-schema`
- Project skeletons: `web-skeleton`, `cli-skeleton`
- Curated list: [awesome-phel](https://github.com/phel-lang/awesome-phel)

**GitHub adoption signal**
- 560 `.phel` files indexed across 25+ unique public repositories (`gh search code extension:phel`)
- This number is likely undercounted: without Linguist support, many user repos go uncategorized and surface poorly in code search, which suppresses further adoption.

Happy to refresh grammar samples or provide any additional info. Thanks for considering!

JesusValeraDev committed:
Merge branch 'main' into add-phel-language

Chemaclass committed:
Merge branch 'main' into add-phel-language

Chemaclass committed:
Merge branch 'main' into add-phel-language
